plugins/serverless/tsconfig.json

```diff
@@ -0,0 +1,13 @@
+{
+  "extends": "../../tsconfig.settings.json",
+  "compilerOptions": {
+    "outDir": "lib",
+    "rootDir": "src"
+  },
+  "references": [
+    {
+      "path": "../../lib/types"
+    }
```

Comment:
**nitpick**: the other Tool Kit packages you use in `serverless` should be included in the `references` object too, e.g., `vault`, `logger`, etc. this ensures that `serverless` is compiled after all of them and means you won't get build errors from trying to import modules that haven't been built yet (though that usually only will happen with a clean build).

Reply:
Ahh interesting, I didn't know that, thanks for the explanation!

Reply:
I've updated that now and I have another question :D - the [Node task includes the options lib](https://github.com/Financial-Times/dotcom-tool-kit/blob/main/plugins/node/tsconfig.json#L18) in its list of references, but I don't see anywhere in the Node plugin where it's directly required - is that just a mistake or am I missing something that means I should potentially be adding other references to my `tsconfig.json` that aren't directly required?

(Also, I see that the `state` lib is also not included in the list of refs for the Node plugin, and that is directly required. I assume that is a mistake, I can make a little PR to add that in...)

Reply:
yep, that is a mistake! we're not very good at maintaining those because usually it doesn't present a problem. it will only cause the first `tsc` build of the project to fail if they're misconfigured and any subsequent runs should work just fine so it's not really a priority. but it is 1) more _correct_ and 2) presents a confusing experience and a bad first impression of the monorepo if your first build fails!

Reply:
I would prefer a hard error if `references` is configured incorrectly tbh but it doesn't look like such an option exists atm. `references` [seems](https://www.typescriptlang.org/docs/handbook/project-references.html#what-is-a-project-reference) like it makes a few more complicated and subtle changes to your project anyway so maybe that's why it's not strictly enforced.